### DISCUSSION TITLE
gcs-cli unable to execute any task like --home, --file etc but is connected. Works on Windows not on Linux

### DISCUSSION THREAD

**Discussion by willemdeschryver** [winder/Universal-G-Code-Sender#2253] [Help]
Hi,

I'm trying to get the gcs-cli running on a raspberrypi zero.
I tested the functionality of the cli interface on Windows using the following command.
This worked perfectly for my gemitsu 4040pro machine using GRBL 1.1h
`java -Xmx256m -cp UniversalGcodeSender.jar com.willwinder.ugs.cli.TerminalClient -h`

When i tried to port this to the RPIzero running raspberry OS I couldn't get it to work.
Connecting isn't the problem but running task like home is.
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/9156ae90-e18c-4c09-a783-542b8950e855)
The same goes for running a file or reseting the alarm
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/13b0e1e6-fb24-4903-b78b-b46e35c5a7f6)

What could be the problem?
Thank you for you help!

Greetings,
Willem

**Comment by breiler** [comment]
For some reason your controller responds with a weird message to the `$G` command:
`GC:G0 G54 G17 G21 G90 G94 M5 M M9 T0 F0 S0`

The single `M` in the response above is faulty. What is the output if you connect to the same CNC controller using a UGS GUI and type the command `$G`?

**Comment by willemdeschryver** [comment]
I'm not around the machine anymore. I got the same error on windows environment aswell, since these both use the same sourcecode i don't know if this will make a difference.
I will test the command tomorrow..
Thank you for your quick response!

**Comment by willemdeschryver** [comment]
I just tested the command $G
`>>> $G
[GC:G0 G54 G17 G21 G90 G94 M5 M M9 T0 F0 S0]
ok`
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/763f9a2e-5069-4574-9d58-e74bc82c8609)


I get the same respons in the GUI.

**Comment by breiler** [comment]
What variant of GRBL are you using (where did you download it)?

What is the response of `$I`?

**Comment by willemdeschryver** [comment]
It's GRBL1.1h
it was preinstalled on the controller of the genmitsu 4040pro
`>>> $I
[VER:GD32 V2.1.20220827:]
[OPT:VZL,35,254]
ok`

**Reply by breiler** [reply]
Ok this is probably the cause. They have changed the version string so that it looks very different than the original variant. UGS can't decipher it and doesn't know what type of controller it is connected to.

On your Windows machine you are probably using an older version of UGS (2.0.15 maybe?) and on your Linux machine you are using a later version.

As I haven't access to their firmware or hardware I will have to guess how to fix this.
I've made an attempt here: https://github.com/winder/Universal-G-Code-Sender/pull/2255

It should now be available in the latest nightly build here:
https://github.com/winder/Universal-G-Code-Sender#downloads

This only fixes the version string which should make it possible to connect to the machine. But we will probably not be able to parse the faulty `$G`-response which could cause other problems...

**Reply by willemdeschryver** [reply]
Hi, i just got the chance to test this out. Im getting the following error using the latest nightly build.
Error is the same when using start-cli.sh
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/d2a3b8c1-1210-410f-8316-6c4a09575149)

**Reply by willemdeschryver** [reply]
I just tested all the previous versions (2.0.15-2.0.17) they both give the same error listed on my previous past. Only 2.0.18 is Running on the Rpi zero w

**Comment by willemdeschryver** [comment]
Running a command on windows give me this.
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/e766f6b9-ce23-4038-bac6-ddb7432d35e9)

it homes the machine without any delay or trouble, it's the same .jar file

**Comment by willemdeschryver** [comment]
Hi, i switched my hardware o a RPI b3 on ths machine i can run the lastet nightly build.
Running the nightly build i get the following error. The machine will start after the error was thrown  but i have the feeling this introduces some delay.
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/14cf9fb5-ba31-451e-a5b0-35a92bf2d663)
This error only throws when running home commands.

When stopping a cli command mind operation and rerunning an other close after i get the following error thrown
![image](https://github.com/winder/Universal-G-Code-Sender/assets/58251215/662c67be-3ebc-45d5-a9d6-e86d94bea7ce)

It seems to me that some responses of the cnc aren't catched.

Thank you for your time !
